index.html

```diff
@@ -413,6 +413,26 @@ <h3>
       </li>
     </ul>
   </p>
+  <p>
+    The <dfn>contributing factors change threshold exceeded</dfn> steps given |factors| and |state|, are as follows:
+    <ol>
+      <li>
+        If |factors| exceed an [=implementation-defined=] threshold for |state|, return true.
```

Comment:
factors is an array right? and they include values as "thermal"... how does that relate to threshold.

I get this if we do it lower in the system, like when we determine the states, because then we shouldn't change the state unless the underlying values differ more than some implementation define value.

Like, if we in an impl decide that 80% is the threshold for changing between fair and serious, then fair->serious should change at 85% maybe and serious->fair at say 75%. I don't see how we can do this, this high up in the algorithm, it's more guidance we could give for when the platform collector changes state

Reply:
I struggled a bit on how to spec this as I tried to understand how the pieces come together in this API. I think this is better and conveys what I actually meant:

If changes in [=contributing factors=] represented by |factors| ...

Reply:
The problem is that contributing factors are not the low level contributing metrics but the exposed factors that helped form the state, ie. its an array of strings like "power-supply" and "thermals" but low level values like CPU utilization in %

Reply:
> If changes in low-level contributing metrics exceed an implementation-defined threshold for state change, return true.

Yes, but it is not always exceeding but could be the opposite if moving from say critical to serious.

Reply:
I guess we could say that state is determined in an implementation defined manner from a set of low-level hardware metrics, metrics that most developers would have a hard time reasoning about (as I have :)) but developers understand high levels factors and that they can contribute to the state, such as thermals (getting too hot) and power supply (not getting enough juice to run at full speed, or use boost modes)